cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1157 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+import warnings
+from typing import List, Dict, Any, Optional, Tuple
+import numpy as np
+import copy
+
+# for MAP
+from multiprocessing import Pool
+
+"""Methods to rank and score images in an object detection dataset (object detection data), based on how likely they
+are to contain label errors. """
+
+
+def get_label_quality_scores(
+    annotations: List[Dict[Any, Any]],
+    results: List[np.ndarray],
+    *,
+    method: str = "map",
+    threshold: Optional[float] = None,
+    verbose: bool = True,
+) -> np.ndarray:
+    """Returns a label quality score for each datapoint.
+
+    This is a function to compute label quality scores for object detection (multi-class) datasets,
+    where lower scores indicate labels less likely to be correct.
+
+    Score is between 0 and 1.
+
+    1 - clean label (given label is likely correct).
+    0 - dirty label (given label is likely incorrect).
+
+    Parameters
+    ----------
+    annotations:
+        A list of dictionaries such that `annotations[i]` contains the given annotations for the `i`-th image in the format
+       `{'bboxes': np.ndarray((N,4)), 'labels': np.ndarray((N,)), 'image_name': str}` where N is the number of bounding boxes
+       for the `i`-th image and `bboxes[j]` is in the format [x,y,w,h] with given label `labels[j]`. ('image_name' is optional here)
+
+    results:
+        A list of np.ndarray where such that `results[i]` corresponds to the predicted results for the `i`-th image
+        in the format `np.ndarray((K,))` where K is the number of classes and `results[i][k]` is of shape `np.ndarray(N,5)`
+        where `N` is the number of bounding boxes for class `K` and the five columns correspond to `[x,y,w,h,pred_prob]` returned
+        by the model.
```

Comment:
Is this the exact same format as output by MMDet models?  If so let's explicitly say so, and link to it. Also link to other object detection frameworks whose predictions come in this format. 

If not, our tutorial will have to show how to convert raw object detection framework outputs into the right format for cleanlab